pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -180,12 +184,14 @@ type resettableRESTMapper interface {
 func (gc *GarbageCollector) Sync(discoveryClient discovery.ServerResourcesInterface, period time.Duration, stopCh <-chan struct{}) {
 	oldResources := make(map[schema.GroupVersionResource]struct{})
 	wait.Until(func() {
+		lastErrorRecord := gc.errorRecorder.DeleteSyncError()
 		// Get the current resource list from discovery.
 		newResources := GetDeletableResources(discoveryClient)
 
 		// This can occur if there is an internal error in GetDeletableResources.
 		if len(newResources) == 0 {
 			klog.V(2).Infof("no resources reported by discovery, skipping garbage collector sync")
+			gc.errorRecorder.SetSyncError(lastErrorRecord.Copy().WithError(goerrors.New("no resources reported by discovery")))
```

Comment:
I have simplified the logic, but we still need the asynchronous maps since we can have multiple workers. So I cannot bring myself to naming it append. Can't we rather use one of the following verbs since we are basically inserting into a map for each node? `set`, `put` or `record` ?